src/include/rados/librados.h

```diff
@@ -931,6 +950,41 @@ CEPH_RADOS_API int rados_nobjects_list_next(rados_list_ctx_t ctx,
  */
 CEPH_RADOS_API void rados_nobjects_list_close(rados_list_ctx_t ctx);
 
+CEPH_RADOS_API rados_enumerate_cursor rados_enumerate_objects_begin(rados_ioctx_t io);
+CEPH_RADOS_API rados_enumerate_cursor rados_enumerate_objects_end(rados_ioctx_t io);
+
+CEPH_RADOS_API int rados_enumerate_objects_is_end(rados_ioctx_t io,
+    rados_enumerate_cursor cur);
+
+CEPH_RADOS_API void rados_enumerate_cursor_free(rados_ioctx_t io,
+    rados_enumerate_cursor cur);
+
+CEPH_RADOS_API int rados_enumerate_cursor_cmp(rados_ioctx_t io,
+    rados_enumerate_cursor lhs, rados_enumerate_cursor rhs);
+
+/**
+ * @return the number of items set in the result array
+ */
+CEPH_RADOS_API int rados_enumerate_objects(rados_ioctx_t io,
+    const rados_enumerate_cursor start,
+    const rados_enumerate_cursor finish,
+    const size_t result_size,
+    rados_enumerate_item *results,
+    rados_enumerate_cursor *next);
+
+CEPH_RADOS_API void rados_enumerate_objects_free(
+    const size_t result_size,
+    rados_enumerate_item *results);
+
+CEPH_RADOS_API void rados_enumerate_objects_split(rados_ioctx_t io,
+    const rados_enumerate_cursor start,
+    const rados_enumerate_cursor finish,
+    const size_t n,
+    const size_t m,
+    rados_enumerate_cursor *split_start,
+    rados_enumerate_cursor *split_finish);
```

Comment:
It took me a minute to remember so docstring definitely needed...

The dual output arguments get you a range from somewhere within the input start/finish, basically (chunk n of m).  Maybe "slice" is a better name than "split" here